sdcflows/interfaces/utils.py

```diff
@@ -147,6 +150,74 @@ def _run_interface(self, runtime):
         return runtime
 
 
+class _ReorientImageAndMetadataInputSpec(TraitedSpec):
+    in_file = File(exists=True, mandatory=True, desc="Input 3- or 4D image")
+    target_orientation = traits.Str(
+        desc="Axis codes of coordinate system to reorient to"
+    )
+    pe_dir = InputMultiObject(
+        traits.Enum(
+            *["".join(p) for p in product("ijkxyz", ("", "-"))],
+            mandatory=True,
+            desc="Phase encoding direction",
+        )
+    )
+
+
+class _ReorientImageAndMetadataOutputSpec(TraitedSpec):
+    out_file = File(desc="Reoriented image")
+    pe_dir = OutputMultiObject(
+        traits.Enum(
+            *["".join(p) for p in product("ijkxyz", ("", "-"))],
+            desc="Phase encoding direction in reoriented image",
+        )
+    )
+
+
+class ReorientImageAndMetadata(SimpleInterface):
+    input_spec = _ReorientImageAndMetadataInputSpec
+    output_spec = _ReorientImageAndMetadataOutputSpec
+
+    def _run_interface(self, runtime):
+        import numpy as np
+        import nibabel as nb
+        from nipype.utils.filemanip import fname_presuffix
+
+        target = self.inputs.target_orientation.upper()
+        if not all(code in "RASLPI" for code in target):
+            raise ValueError(
+                "Invalid orientation code {self.inputs.target_orientation}"
+            )
+
+        img = nb.load(self.inputs.in_file)
+        img2target = nb.orientations.ornt_transform(
+            nb.io_orientation(img.affine),
+            nb.orientations.axcodes2ornt(self.inputs.target_orientation),
+        ).astype(int)
+
+        # Identity transform
+        if np.array_equal(img2target, [[0, 1], [1, 1], [2, 1]]):
+            self._results = dict(
+                out_file=self.inputs.in_file,
+                pe_dir=self.inputs.pe_dir,
+            )
```

Comment:
Should we just exit here?
```suggestion
            )
            return runtime
```